server/concept/ConceptService.java

```diff
@@ -28,69 +28,96 @@
-import static com.vaticle.typedb.core.server.common.ResponseBuilder.ConceptManager.getThingRes;
 import static com.vaticle.typedb.core.server.common.ResponseBuilder.ConceptManager.putAttributeTypeRes;
 import static com.vaticle.typedb.core.server.common.ResponseBuilder.ConceptManager.putEntityTypeRes;
 import static com.vaticle.typedb.core.server.common.ResponseBuilder.ConceptManager.putRelationTypeRes;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.GET_ATTRIBUTE_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.GET_ATTRIBUTE_TYPE_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.GET_ENTITY_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.GET_ENTITY_TYPE_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.GET_RELATION_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.GET_RELATION_TYPE_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.GET_SCHEMA_EXCEPTIONS_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.PUT_ATTRIBUTE_TYPE_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.PUT_ENTITY_TYPE_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.PUT_RELATION_TYPE_REQ;
+import static com.vaticle.typedb.protocol.ConceptProto.ConceptManager.Req.ReqCase.REQ_NOT_SET;
+
 
 public class ConceptService {
 
     private final TransactionService transactionSvc;
     private final ConceptManager conceptMgr;
 
+    private final EnumMap<ConceptProto.ConceptManager.Req.ReqCase, BiConsumer<ConceptProto.ConceptManager.Req, UUID>> handlers;
+
     public ConceptService(TransactionService transactionSvc, ConceptManager conceptMgr) {
         this.transactionSvc = transactionSvc;
         this.conceptMgr = conceptMgr;
+
+        handlers = new EnumMap<>(ConceptProto.ConceptManager.Req.ReqCase.class);
+        handlers.put(GET_ENTITY_TYPE_REQ, this::getEntityType);
+        handlers.put(GET_RELATION_TYPE_REQ, this::getRelationType);
+        handlers.put(GET_ATTRIBUTE_TYPE_REQ, this::getAttributeType);
+        handlers.put(PUT_ENTITY_TYPE_REQ, this::putEntityType);
+        handlers.put(PUT_ATTRIBUTE_TYPE_REQ, this::putAttributeType);
+        handlers.put(PUT_RELATION_TYPE_REQ, this::putRelationType);
+        handlers.put(GET_ENTITY_REQ, this::getEntity);
+        handlers.put(GET_RELATION_REQ, this::getRelation);
+        handlers.put(GET_ATTRIBUTE_REQ, this::getAttribute);
+        handlers.put(GET_SCHEMA_EXCEPTIONS_REQ, this::getSchemaExceptions);
+        handlers.put(REQ_NOT_SET, this::requestNotSet);
+        assert handlers.size() == ConceptProto.ConceptManager.Req.ReqCase.class.getEnumConstants().length;
```

Comment:
Using an `EnumMap` for the request handlers forces them to have uniform API and cleans up the `execute()` internals. The `assert`ion guards against us forgetting to implement a request.